.github/scripts/gitutils.py

```diff
@@ -212,11 +217,21 @@ def cherry_pick_commits(self, from_branch: str, to_branch: str) -> None:
             self.cherry_pick(commit)
         self.checkout(orig_branch)
 
-    def push(self, branch: str, dry_run: bool) -> None:
-        if dry_run:
-            self._run_git("push", "--dry-run", self.remote, branch)
-        else:
-            self._run_git("push", self.remote, branch)
+    def push(self, branch: str, dry_run: bool, retry: int = 3) -> None:
+        for cnt in range(retry):
+            try:
+                if dry_run:
+                    self._run_git("push", "--dry-run", self.remote, branch)
+                else:
+                    self._run_git("push", self.remote, branch)
+            except RuntimeError as e:
+                # Check if push were rejected because branch is stale
+                if len(e.args) != 1:
+                    raise
```

Comment:
This check is kinda sketchy. Is there a better way to do this?

Reply:
This limits the scope on what kinds of Exceptions  we are attempting to retry on (i.e. if RuntimeError is raised somewhere else in the stack, but I can probably fold those two checks into one